CMSgov/design-system
# [WNMGDS-966] & [WNMGDS-949] - updating typescript definitions for Badge & Alert

## Summary
When talking to @nichia, I realized I missed some typescript definition things in the Badge and Alert components. These changes are pulled from the original typescript type files.

### Changed
Updated Badge and Alert component typescript definitions to better match the originals defined in their 'type' files.

- [original Alert type for reference](https://github.com/CMSgov/design-system/blob/354cd70ef0f69e235459a705a8637d76e5638a54/packages/design-system/src/types/Alert/Alert.d.ts)
- [original Badge type for reference](https://github.com/CMSgov/design-system/blob/a1dd4d99c37f6fa807377f20ff9314407ab99eb4/packages/design-system/src/types/Badge/Badge.d.ts)

## How to test
Run tests, ensure that all tests pass
Run doc site and ensure that there aren't any build errors
